packages/back-end/src/models/SlackIntegrationModel.ts

```diff
@@ -194,6 +194,122 @@ export const getSlackIntegration = async ({
   }
 };
 
+type GetForEventOptions = {
+  organizationId: string;
+  eventName: NotificationEventName;
+  environments: string[];
+  tags: string[];
+  projects: string[];
+};
+
+/**
+ * Filters by the following:
+ *  eventName:
+ *    If the integration's events includes the provided event, or
+ *    if the integration does not specify events,
+ *    it will be included.
+ *  environments:
+ *    If the integration's environments intersects with the integration's environments, or
+ *    if the integration does not specify environments,
+ *    it will be included.
+ *  tags:
+ *    If the integration's tags intersects with the integration's tags, or
+ *    if the integration does not specify tags,
+ *    it will be included.
+ *  projects:
+ *    If the integration's projects intersects with the integration's projects, or
+ *    if the integration does not specify projects,
+ *    it will be included.
+ * @param organizationId
+ * @param eventName
+ * @param environments
+ * @param tags
+ * @param projects
+ */
+export const getSlackIntegrationsForFilters = async ({
+  organizationId,
+  eventName,
+  environments,
+  tags,
+  projects,
+}: GetForEventOptions): Promise<SlackIntegrationInterface[] | null> => {
+  try {
+    const docs = await SlackIntegrationModel.find({
```

Comment:
I think in this case, it would be better to get all slack integrations for an organization and do the filtering in code.  That way, every event for an org will execute the exact same Mongo query which is better for optimization, makes it easier to cache in the future, and makes the code much easier to read and maintain.

Reply:
I've made the change to query for all SlackIntegration records for an organization in Mongo up front, then applied filtering in code. You can see the changes here: 5426323a8347ba86cda6ad9033a829f47d9a9724